contracts/provider/vault/src/contract.rs

```diff
@@ -437,6 +451,146 @@ impl VaultContract<'_> {
         Ok(())
     }
 
+    /// Updates the pending txs for remote staking on any contract
+    ///
+    /// Stake (remote) is always called by the tokens owner, so the `sender` is
+    /// used as an owner address.
+    ///
+    /// Config is taken in argument as it sometimes is used outside of this function, so
+    /// we want to avoid double-fetching it
+    fn maybe_stake(
+        &self,
+        ctx: &mut ExecCtx,
+        config: &Config,
+        lienholder: &Addr,
+        slashable: Decimal,
+        amount: Coin,
+    ) -> Result<u64, ContractError> {
+        ensure!(
+            amount.denom == config.denom,
+            ContractError::UnexpectedDenom(config.denom.clone())
+        );
+
+        let amount = amount.amount;
+        // Tx starts here
+        // Verify that the user has enough collateral to stake this and the currently pending txs
+        let pending_amount = amount
+            + self
+                .pending
+                .txs
+                .idx
+                .users
+                .prefix(ctx.info.sender.clone())
+                .range(ctx.deps.storage, None, None, Order::Ascending)
+                .fold(Ok(Uint128::zero()), |acc, pending| {
+                    let acc = acc?;
+                    pending.map(|(_, tx)| {
+                        acc + match tx.ty {
+                            // Value range max
+                            TxType::Stake => tx.amount,
+                            _ => Uint128::zero(),
+                        }
+                    })
+                })?;
+
+        // Load lien and update (but do not save) user info and lien holder
+        let lien = self
+            .liens
+            .may_load(ctx.deps.storage, (&ctx.info.sender, lienholder))?
+            .unwrap_or(Lien {
+                amount: Uint128::zero(),
+                slashable,
+            });
+        // Load user and update (but do not save) max lien and total slashable
+        let mut user = self
+            .users
+            .may_load(ctx.deps.storage, &ctx.info.sender)?
+            .unwrap_or_default();
+        user.max_lien = user.max_lien.max(lien.amount);
```

Comment:
It's just the sum of the pending ones:
```suggestion
        user.max_lien = user.max_lien.max(peding_amount);
```